lib/exercism/github.rb

```diff
@@ -21,18 +21,23 @@ def self.authenticate(code, client_id, client_secret)
       req.headers['User-Agent'] = user_agent
       req.body = { client_id: client_id, client_secret: client_secret, code: code }.to_json
     end
-    access_token = JSON.parse(response.body)['access_token']
+    JSON.parse(response.body)['access_token']
+  end
 
+  def self.user_info(access_token)
     conn = connect_to 'https://api.github.com'
     response = conn.get do |req|
       req.url '/user'
       req.headers['User-Agent'] = user_agent
       req.params['access_token'] = access_token
     end
-    result = JSON.parse(response.body)
-    [result['id'], result['login'], result['email'], result['avatar_url']]
+    if response.status != 200
+      raise UnauthorizedUser
+    else
+      result = JSON.parse(response.body)
+      [result['id'], result['login'], result['email'], result['avatar_url']]
```

Comment:
We're passing this set of 4 parameters around all over the place. Now that we're using these in more places, the next refactoring step should be [introduce parameter object]. I think `GithubUser` could be a reasonable abstraction, with these 4 fields, and a constructor `GithubUser.from_response(response)`. Then `User#from_github` can take a `GithubUser` instead of the 4 parameters it currently requires. 

[introduce parameter object]: https://sourcemaking.com/refactoring/introduce-parameter-object